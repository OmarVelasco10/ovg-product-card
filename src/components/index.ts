import { ProductCard as ProductCardHOC } from "./ProductCard";

import { ProductButtons } from "./ProductButtons";
import { ProductImage } from "./ProductImgae";
import { ProductTitle } from "./ProductTitle";
import { ProductCardHOCProps } from "../interfaces/interfaces";

export { ProductButtons } from "./ProductButtons";
export { ProductImage } from "./ProductImgae";
export { ProductTitle } from "./ProductTitle";

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
})

export default ProductCard;

