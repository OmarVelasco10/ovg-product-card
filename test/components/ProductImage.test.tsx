import renderer from 'react-test-renderer';
import * as React from 'react';
// import * as ReactDOM from 'react-dom';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/product';

describe('ProductImage test', () => {
    it('should to show the component with the custom image', () => {
      const wrapper = renderer.create(
        <ProductImage img="https://test.jpg" />
      );

      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('Should to show the component with the product image', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
          );
    
          expect(wrapper.toJSON()).toMatchSnapshot();
    })
  });
  