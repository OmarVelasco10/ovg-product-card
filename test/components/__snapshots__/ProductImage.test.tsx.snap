// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage test Should to show the component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage test should to show the component with the custom image 1`] = `
<img
  alt="Product img"
  className="productImg undefined"
  src="https://test.jpg"
/>
`;
